README.md
# SMAT(Smart Mat) (init 01.25.19)
# Done: 01.25.2019. - 08.23.2019.--Silver Prize--

Final Product Top 3
## I-Caps
The International Students Joint Capstone Design Project (i-CAPS).


The i-CAPS program and the  which include International Students Multidisciplinary Design Camps (d-CAMP) is a collaboration program involving 
* 18 Universities in 6 Countries
  * Institut Teknologi Bandung (Indonesia)
  * Universiti Kebangsaan Malaysia (Malaysia)
  * Nanyang Technological University (Singapore)
  * Tianjin University (China)
  * National Cheng Kung University (Taiwan)
  * Chonbuk National University (Korea - HICEE)
  * A consortium of 13 Korean universities(headquartered in Chonbuk National University)

----

## SMART Living

> For Ordinary Person & Blind Person

* Arduino Project
  * Mat
  * LCD
  * RSS - Weather Agency
  
* Raspberry Pi Project (01.31.19. 19:00)
  * > Consider switching to raspberry pi


* Product Description

![team11_final smart_ma t](https://user-images.githubusercontent.com/43804152/51780706-95bd8480-214b-11e9-9085-e46eba5f6cbb.jpg)
![team11_final smart_mat](https://user-images.githubusercontent.com/43804152/51780705-9524ee00-214b-11e9-97ee-683dabbf3e86.jpg)

* Schedule
  * 1st D.Camp : 01.20.19 - 01.25.19, NCKU, Tainan, Taiwan, `Off-line`
  * 1st Presentation : 01.26.19, NCKU, Tainan, Taiwan, `Off-line`
  * 2nd Presentation : 04.12.19, `On-line` (Team 11 Presenter: Luqman Hakimi)
  * 3rd Presentation : 05.23.19, `On-line` -> 02-03.07.19
  * Deadline : 08.21.19, `Product`
  * 2nd D.Camp : 08.22.19 - 08.26.19, kt human resource development center, Daejeon, South Korea, `Off-line`
  * Implementation : 04.14.19.

* [Work Flow](https://github.com/jeongyoonlee2015/icaps-icaps-d.camp/blob/master/Plan/00.Work_Flow.md)  
    
* Feedback
  * 1st Presentation
   1.	Maybe can have some personal schedule in the display panel
   2.	Display the weight and keep the data recorded for the future reference
   3.	Adding a foot print sensor into the floor mat to indicate the standing posture of the user
   
   
  * 2nd Presentation
   1. LRC and WBS were impressive.
  
  
  * Next Goals: 2nd Presentation (1st interim video conference) was process oriented presentation.
   3nd Presentation( 2nd interim video conference) will be implementation product oriented presentation.


* Reference
  * [Electronic Circuit Simulator](https://www.multisim.com/)
  * [Weather RSS](http://www.weather.go.kr/weather/lifenindustry/sevice_rss.jsp)
  * [Similar Product](https://www.geeky-gadgets.com/smartmat-the-connected-door-mat-25-03-2014/)
  * [Kocoafab Tutorial](https://kocoafab.cc/tutorial/view/595)
  * [Circuit Simulating](https://www.tinkercad.com/things/6VZ8ewfz5qE-start-simulating/editel?collectionid=OIYJ88OJ3OPN3EA&lessonid=EHD2303J3YPUS5Z&magic=fea420dba067cf0cd52378305ce02fb0d680515b&projectid=OIYJ88OJ3OPN3EA&t=1548925471689927833&tenant=circuits#/lesson-viewer)
  
  
> Another Resource <br> [HackerShack git](https://github.com/HackerShackOfficial/Smart-Mirror) <br> [DHS연구소](https://dhslab.modoo.at/)


news.md
```
{
			module: "newsfeed",
			position: "bottom_bar",
			config: {
				feeds: [
					{
						title: "My news",
						url: "http://www.nytimes.com/services/xml/rss/nyt/HomePage.xml"
						//url: "http://rss.ohmynews.com/rss/ab_reporter.xml"안 됨
					}
				],
				showSourceTitle: true,
				showPublishDate: true,
				broadcastNewsFeeds: true,
				broadcastNewsUpdates: true
			}
		},
```


JsonConfig/Mich.js
/* Magic Mirror Config Sample
 *
 * By Michael Teeuw http://michaelteeuw.nl
 * MIT Licensed.
 *
 * For more information how you can configurate this file
 * See https://github.com/MichMich/MagicMirror#configuration
 *
 * --완벽한 원본은 아님--
 */

var config = {
	address: "localhost", // Address to listen on, can be:
	                      // - "localhost", "127.0.0.1", "::1" to listen on loopback interface
	                      // - another specific IPv4/6 to listen on a specific interface
	                      // - "", "0.0.0.0", "::" to listen on any interface
	                      // Default, when address config is left out, is "localhost"
	port: 8080,
	ipWhitelist: ["127.0.0.1", "::ffff:127.0.0.1", "::1"], // Set [] to allow all IP addresses
	                                                       // or add a specific IPv4 of 192.168.1.5 :
	                                                       // ["127.0.0.1", "::ffff:127.0.0.1", "::1", "::ffff:192.168.1.5"],
	                                                       // or IPv4 range of 192.168.3.0 --> 192.168.3.15 use CIDR format :
	                                                       // ["127.0.0.1", "::ffff:127.0.0.1", "::1", "::ffff:192.168.3.0/28"],

	language: "en",
	timeFormat: 24,
	units: "metric",

	modules: [
		{
			module: "alert",
		},
		{
			module: "updatenotification",
			position: "top_bar"
		},
		{
			module: "clock",
			position: "top_left"
		},
		{
			module: "calendar",
			header: "SMAT",
			position: "top_left",
			config: {
				calendars: [
					{
						symbol: "calendar-check",
						url: "https://calendar.google.com/calendar/ical/2019captain11%40gmail.com/private-54ccbdf0a89e8f20a3dc94d991308ef5/basic.ics"
					}
				]
			}
		},
		{
			module: "compliments",
			position: "lower_third"
		},
		{
			module: "currentweather",
			position: "top_right",
			config: {
				location: "Seoul",
				locationID: "",  //ID from http://bulk.openweathermap.org/sample/city.list.json.gz; unzip the gz file and find your city
				appid: ".."
			}
		},		{
			module: "weatherforecast",
			position: "top_right",
			header: "Weather Forecast",
			config: {
				location: "Seoul",
				locationID: "1835847",  //ID from http://bulk.openweathermap.org/sample/city.list.json.gz; unzip the gz file and find your city
				appid: ".."
			}
		},
		{
			module: "newsfeed",
			position: "bottom_bar",
			config: {
				feeds: [
					{
						title: "New York Times",
						url: "http://www.nytimes.com/services/xml/rss/nyt/HomePage.xml"
					}
				],
				showSourceTitle: true,
				showPublishDate: true,
				broadcastNewsFeeds: true,
				broadcastNewsUpdates: true
			}
		},
	]

};

/*************** DO NOT EDIT THE LINE BELOW ***************/
if (typeof module !== "undefined") {module.exports = config;}


JsonConfig/config.js
/* Magic Mirror Config Sample
 *
 * By Michael Teeuw http://michaelteeuw.nl
 * MIT Licensed.
 *
 * For more information how you can configurate this file
 * See https://github.com/MichMich/MagicMirror#configuration
 *
 */

var config = {
	address: "localhost", // Address to listen on, can be:
	                      // - "localhost", "127.0.0.1", "::1" to listen on loopback interface
	                      // - another specific IPv4/6 to listen on a specific interface
	                      // - "", "0.0.0.0", "::" to listen on any interface
	                      // Default, when address config is left out, is "localhost"
	port: 8080,
	ipWhitelist: ["127.0.0.1", "::ffff:127.0.0.1", "::1"], // Set [] to allow all IP addresses
	                                                       // or add a specific IPv4 of 192.168.1.5 :
	                                                       // ["127.0.0.1", "::ffff:127.0.0.1", "::1", "::ffff:192.168.1.5"],
	                                                       // or IPv4 range of 192.168.3.0 --> 192.168.3.15 use CIDR format :
	                                                       // ["127.0.0.1", "::ffff:127.0.0.1", "::1", "::ffff:192.168.3.0/28"],

	language: "en",
	timeFormat: 24,
	units: "metric",

	modules: [
		{
			module: "alert",
		},
		{
			module: "updatenotification",
			position: "top_bar"
		},
		{
			module: "clock",
			position: "top_left"
		},
		{
			module: "calendar",
			header: "SMAT",
			position: "top_left",
			config: {
				calendars: [
					{
						symbol: "calendar-check",
						url: ""
					}
				]
			}
		},
		{
			module: "compliments",
			position: "lower_third",
			config: {
			  compliments: {
				anytime:[
					"Hello, Captain11!",
					"UKM X CBNU SMAT"
				],
				morning:[
					"Hello,Captain",
					"Selamat pagi",
					"Good morning Captain"
				],
				afternoon:[
					  "Nice Captain",
					  "Selamat petan"
				]
			  }
			}
		},
		{
			module: "currentweather",
			position: "top_right",
			config: {
				location: "Seoul",
				locationID: "",  //ID from http://bulk.openweathermap.org/sample/city.list.json.gz; unzip the gz file and find your city
				appid: ""
			}
		},		{
			module: "weatherforecast",
			position: "top_right",
			header: "Weather Forecast",
			config: {
				location: "Seoul",
				locationID: "1835847",  //ID from http://bulk.openweathermap.org/sample/city.list.json.gz; unzip the gz file and find your city
				appid: ""
			}
		},
		{
			module: "newsfeed",
			position: "bottom_bar",
			config: {
				feeds: [
					{
						title: "New York Times",
						url: "http://www.nytimes.com/services/xml/rss/nyt/HomePage.xml"
					}
				],
				showSourceTitle: true,
				showPublishDate: true,
				broadcastNewsFeeds: true,
				broadcastNewsUpdates: true
			}
		},
		{
		      module: "MMM-AssistantMk2",
  		      position: "top_right",
      		config: {
        	  record: {
                    recordProgram : "arecord",  
          	    device        : "plughw:1",
        	},
        	notifications: {
          	  ASSISTANT_ACTIVATED: "HOTWORD_PAUSE",
          	  ASSISTANT_DEACTIVATED: "HOTWORD_RESUME",
        	},
 	        deviceLocation: {
         	  coordinates: {
            	    latitude: 37.57,
                    longitude: 126.98
          	   }
                 },
        	profiles: {
          	  "default": {
           	   lang: "ko-KR"
          	  }
 	         }
      		}
	      },
     	    {
      	     module: "MMM-Hotword",
      	     config: {
        	snowboy: [
          	 {
	            hotwords: "smartmirror",
        	    file: "resources/models/smart_mirror.umdl",
            	    sensitivity: '1.0',
                 },
          	 {
   	          hotwords: "snowboy",
                  file: "resources/models/snowboy.umdl",
            	  sensitivity: '1.0',
          	 },
           	 {
	            file: 'resources/models/jarvis.umdl',
        	    sensitivity: '0.8,0.80',
            	    hotwords: ['jarvis', 'jarvis']
         	 }
	        ],
        	 record: {
          	   recordProgram : "arecord",  
     	           device        : "plughw:1",
        	 },
	         autostart:true,
        	 onDetected: {
          	   notification: function (payload) {
            		return "ASSISTANT_ACTIVATE"
          	   },
        	   payload: function (payload){
            	     return {
              		profile: payload.hotword
            	       }
	              }
        	    },
	          },
	        },
	]

};

/*************** DO NOT EDIT THE LINE BELOW ***************/
if (typeof module !== "undefined") {module.exports = config;}


JsonConfig/simpteam.js
var config = {
  address: "localhost",
  port: 8080,
  ipWhitelist: ["127.0.0.1", "::ffff:127.0.0.1", "::1"],
  language: "ko",
  timeFormat: 12,
  units: "metric",

  modules: [
    {
      module: "alert",
    },
    {
      module: "updatenotification",
      position: "top_bar"
    },
    {
      module: "clock",
      position: "top_left",
      config: {
        dateFormat: "LL dddd",
      }
    },
    {
      module: "calendar",
      position: "top_left",
      config: {
        urgency: 21,
        fetchInterval: 3600000,
        calendars: [{
          url: 'https://calendar.google.com/calendar/ical/ko.south_korea%23holiday%40group.v.calendar.google.com/public/basic.ics',
          symbol: '대한민국 공휴일',
        }, ]
      }
    },
    {
      module: "currentweather",
      position: "top_right",
      config: {
        location: "Seoul, KR", //https://openweathermap.org/ 에서 지역을 찾으세요.
        locationID: "",
        appid: "" //openweathermap.org의 API key를 입력하세요.
      }
    },
    {
      module: "weatherforecast",
      position: "top_right",
      config: {
        location: "Seoul, KR", //https://openweathermap.org/ 에서 지역을 찾으세요.
        locationID: "",
        appid: "" //openweathermap.org의 API key를 입력하세요.
      }
    },
    {
      module: "compliments",
      position: "lower_third",
      config: {
        compliments: {
          anytime: [ //아무때나
            "오늘도 좋은 하루!"
          ],
          morning: [ //아침
            "좋은 아침!",
            "힘찬 아침!",
            "잘 잤나요?"
          ],
          afternoon: [ //오후
            "안녕하세요!",
            "멋져요!",
            "잘 지내고 있나요!"
          ],
          evening: [ //저녁
            "와우! 잘 지냈나요?",
            "멋져보이네요!",
            "반가워요!"
          ],
          day_sunny: [//맑은 낮
            
          ],
          day_cloudy: [//흐린 낮
            
          ],
          cloudy: [//흐림

          ],
          cloudy_windy: [//흐리고 바람

          ],
          showers: [//소나기

          ],
          rain: [//비

          ],
          thunderstorm: [//천둥번개

          ],
          snow: [//눈

          ],
          fog: [//안개

          ],
          night_clear: [//맑은 밤

          ],
          night_cloudy: [//흐린 밤

          ],
          night_showers: [//소나기 밤

          ],
          night_rain: [//비오는 밤

          ],
          night_thunderstorm: [//천둥번개 밤

          ],
          night_snow: [//눈오는 밤

          ],
          night_alt_cloudy_windy: [//흐리고 바람부는 밤

          ],
        }
      }
    },
    {
      module: "MMM-AssistantMk2",
      position: "top_right",
      config: {
        record: {
          recordProgram : "arecord",  
          device        : "plughw:1",
        },
        notifications: {
          ASSISTANT_ACTIVATED: "HOTWORD_PAUSE",
          ASSISTANT_DEACTIVATED: "HOTWORD_RESUME",
        },
        deviceLocation: {
          coordinates: {
            latitude: 37.57,
            longitude: 126.98
          }
        },
        profiles: {
          "default": {
           lang: "ko-KR"
          }
        }
      }
    },
    {
      module: "MMM-Hotword",
      config: {
        snowboy: [
          {
            hotwords: "smartmirror",
            file: "resources/models/smart_mirror.umdl",
            sensitivity: '1.0',
          },
          {
            hotwords: "snowboy",
            file: "resources/models/snowboy.umdl",
            sensitivity: '1.0',
          },
          {
            file: 'resources/models/jarvis.umdl',
            sensitivity: '0.8,0.80',
            hotwords: ['jarvis', 'jarvis']
          }
        ],
        record: {
          recordProgram : "arecord",  
          device        : "plughw:1",
        },
        autostart:true,
        onDetected: {
          notification: function (payload) {
            return "ASSISTANT_ACTIVATE"
          },
          payload: function (payload){
            return {
              profile: payload.hotword
            }
          }
        },
      },
    },
  ]
};

/*************** DO NOT EDIT THE LINE BELOW ***************/
if (typeof module !== "undefined") {
  module.exports = config;
}


JsonConfig/Recent.js
/* Magic Mirror Config Sample
 *
 * By Michael Teeuw http://michaelteeuw.nl
 * MIT Licensed.
 *
 * For more information how you can configurate this file
 * See https://github.com/MichMich/MagicMirror#configuration
 *
 */

var config = {
	address: "localhost", // Address to listen on, can be:
	                      // - "localhost", "127.0.0.1", "::1" to listen on loopback interface
	                      // - another specific IPv4/6 to listen on a specific interface
	                      // - "", "0.0.0.0", "::" to listen on any interface
	                      // Default, when address config is left out, is "localhost"
	port: 8080,
	ipWhitelist: ["127.0.0.1", "::ffff:127.0.0.1", "::1"], // Set [] to allow all IP addresses
	                                                       // or add a specific IPv4 of 192.168.1.5 :
	                                                       // ["127.0.0.1", "::ffff:127.0.0.1", "::1", "::ffff:192.168.1.5"],
	                                                       // or IPv4 range of 192.168.3.0 --> 192.168.3.15 use CIDR format :
	                                                       // ["127.0.0.1", "::ffff:127.0.0.1", "::1", "::ffff:192.168.3.0/28"],

	language: "en",
	timeFormat: 24,
	units: "metric",

	modules: [
		{
			module: "alert",
		},
		{
			module: "updatenotification",
			position: "top_bar"
		},
		{
			module: "clock",
			position: "top_left"
		},
		{
			module: "calendar",
			header: "SMAT",
			position: "top_left",
			config: {
				calendars: [
					{
						symbol: "calendar-check",
						url: "https://calendar.google.com/calendar/ical/2019captain11%40gmail.com/private-54ccbdf0a89e8f20a3dc94d991308ef5/basic.ics"
					}
				]
			}
		},
		{
			module:"compliments",
			position: "top_left",
			config:{
				compliments: {
					anytime:[
						"HEIGT: 175cm"
					]
				}
			}
		},
		{
			module: "compliments",
			position: "lower_third",
			config: {
			  compliments: {
				anytime:[
					"UKM X CBNU SMAT",
					"Hello, Captain11!"
				],
				morning:[
					"Hello,Captain",
					"Selamat pagi",
					"Good morning Captain"
				],
				afternoon:[
					  "Nice Captain",
					  "Selamat petan"
				]
			  }
			}
		},

    {
      module: "MMM-DarkSkyForecast",
      header: "Weather Forecast: Daejeon",
      position: "top_right",
      classes: "default everyone",
      disabled: false,
      config: {
        apikey: "a2a06f1b519edad7ea7950f0d00b6e5d",
        latitude: "36.330101",
        longitude: "127.422470",
        iconset: "4c",
        concise: false,
        forecastLayout: "table"
      }
    },
		{
			module: 'MMM-GoogleFit',
			position: "top_left",
			config: {
			}
		},
			{
	    disabled: false,
	    module: "MMM-BMI",
	    position: "top_left",
	    config: {
	        name: "Captain",                    // Your name
	        imperialUnits: true,		      // true or (false = metric)
	        weight: "155",                    // lbs and inches or kg and m
	        height: "71",                     // lbs and inches or kg and m
	        herName: "SMAT",
	        herWeight: "100",                 // lbs and inches or kg and m
	        herHeight: "65",                 // lbs and inches or kg and m
	        useHeader: true,                 // true if you want a header
	        header: "YOUR BODY MASS INDEX",
	        maxWidth: "350px",
	    }
		},
		{
			module: 'MMM-AirQuality',
			position: 'top_right', // you may choose any location
			config: {
	  		location: 'daejeon' // the location to check the index for
				}
		},
		
		{
			module: "newsfeed",
			position: "bottom_bar",
			config: {
				feeds: [
					{
						title: "oh my news",
						url: "http://www.nytimes.com/services/xml/rss/nyt/HomePage.xml"
						//url: "http://rss.ohmynews.com/rss/ab_reporter.xml"안 됨
					}
				],
				showSourceTitle: true,
				showPublishDate: true,
				broadcastNewsFeeds: true,
				broadcastNewsUpdates: true
			}
		},
	]

};

/*************** DO NOT EDIT THE LINE BELOW ***************/
if (typeof module !== "undefined") {module.exports = config;}
